Task 2.1: ExpenseService

evg-dragalev opened the pull request:
Implement ExpenseService singleton and move hardcoded Expenses from ExpenseListServlet

evg-dragalev committed:
Task 2.1: ExpenseService

evg-dragalev committed:
Task 2.1: ExpenseService

Renamed service, added ExpenseService interface

andrewosipenko commented on src/main/java/com/es/jointexpensetracker/service/ExpenseServiceSingleton.java:
```diff
@@ -0,0 +1,51 @@
+                new Expense(14L, "Bus tickets from Warsaw to Minsk", new BigDecimal(200), "Andrei", expenseGroup)
+        );
+    }
+
```
Add @Override

andrewosipenko commented on src/main/java/com/es/jointexpensetracker/service/ExpenseServiceSingleton.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.List;
+import java.util.UUID;
+
+public class ExpenseServiceSingleton implements ExpenseService {
```
The class is not thread safe. Can be called from multiple HTTP threads

evg-dragalev committed:
Task 2.1: ExpenseService

@Override added, thread safety implemented.

evg-dragalev committed:
Task 2.3: View expense details 

Add expense.jsp, implement logic to load expense details.

evg-dragalev committed:
Task 2.3: View expense details

Add expense.jsp, implement logic to load expense details.

evg-dragalev committed:
Task 2.3: View expense details

implemented loadExpenseById() in ExpenseService, add exception handling in ExpenseServlet

evg-dragalev committed:
Task 2.4: Update expense details

Add neсessary logic

evg-dragalev committed:
Merge branch 'develop'

evg-dragalev committed:
Cleaning code

evg-dragalev committed:
Task 2.5: Expense update message

Added filter for serving notification messages.

evg-dragalev committed:
Task 2.5: Expense update message

Merge branch 'develop'

evg-dragalev committed:
Task 2.6: Delete expense

Implement expense deleting

evg-dragalev committed:
Task 2.6: Delete expense

Merge branch 'develop'

evg-dragalev committed:
Task 2.7: Create new expense

Add .jsp for "Add expense" page

evg-dragalev committed:
Task 2.7: Create new expense

Modify ExpenseService and ExpenseServlet for expense creating

evg-dragalev committed:
Expense page prettyfing

evg-dragalev committed:
Merge branch 'develop'

evg-dragalev committed:
Add simple validation to forms

evg-dragalev committed:
Notifications implementation refactoring

NotificationsService created
Session creating error fixed

evg-dragalev committed:
Project structure refactoring

/filter moved into /web

evg-dragalev committed:
Code refactoring

Exceptions handling refactoring (add .jsp and servlet for exceptions handling).
Minor ExpenseServlet refactoring.

evg-dragalev committed:
Task 2.8: Expense pie chart

Create StatisticsService and StatisticsServlet

evg-dragalev committed:
Merge branch 'develop'

evg-dragalev committed:
Task 2.8: Expense pie chart

Map baked by LinkedHashMap now

evg-dragalev committed:
Task 2.9: Expense debts

evg-dragalev committed:
Merge branch 'develop'

evg-dragalev committed:
Task 2.10: Add expense group

evg-dragalev committed:
Error page refactoring

evg-dragalev committed:
Task 2.10: Add expense group

Modify index.jsp and ExpenseServiceServlet, for redirection to act properly

evg-dragalev committed:
Merge branch 'develop'

evg-dragalev committed:
Task 2.11: Add home page

evg-dragalev committed:
Code refactoring

Pie chart names transformation into array done in .jsp now
ExpenseKey class moved to separate file.

evg-dragalev committed:
Merge branch 'develop'
